core/src/main/java/org/springframework/security/access/expression/ExpressionUtils.java

```diff
@@ -30,4 +31,14 @@ public static boolean evaluateAsBoolean(Expression expr, EvaluationContext ctx)
 					+ expr.getExpressionString() + "'", e);
 		}
 	}
+
+	public static Mono<Boolean> evaluateAsMonoBoolean(Expression expr, EvaluationContext ctx) {
```

Comment:
Adding a Reactive return type to a class used by non-reactive applications can potentially break non-reactive applications since depending on the JDK it may load all method definitions as it loads the class. In general, optional dependencies must be placed in distinct classes.